apps/api/src/app/partner-integrations/usecases/update-vercel-configuration/update-vercel-configuration.usecase.ts

```diff
@@ -0,0 +1,261 @@
+          teamId: configurationDetails.teamId,
+          token: configurationDetails.accessToken,
+        });
+      }
+
+      await this.saveProjectIds(command);
+
+      return {
+        success: true,
+      };
+    } catch (error) {
+      throw new ApiException(error.message);
+    }
+  }
+
+  private async getEnvKeys(organizationIds: string[]): Promise<EnvironmentEntity[]> {
+    return await this.environmentRepository.find(
+      {
+        _organizationId: organizationIds,
+      },
+      'apiKeys identifier _organizationId'
+    );
+  }
+
+  private mapProjectKeys(
+    envData: EnvironmentEntity[],
+    projectData: Record<string, string[]>,
+    projectDetails: MappedProjectData
+  ) {
+    const { addProjectIds, updateProjectDetails } = projectDetails;
+    const mappedData = envData.reduce<Record<string, MapProjectkeys>>((acc, curr) => {
+      const projectIds = projectData[curr._organizationId];
+      const newData = {
+        privateKey: curr.apiKeys[0].key,
+        clientKey: curr.identifier,
+        updateProjectDetails: updateProjectDetails.filter((detail) => projectIds.includes(detail.projectId)),
+        addProjectIds: projectIds.filter((id) => addProjectIds.includes(id)),
+      };
+      acc[curr._organizationId] = newData;
+
+      return acc;
+    }, {});
+
+    return mappedData;
+  }
+
+  private async setEnvironments({ clientKey, projectIds, privateKey, teamId, token }: ISetEnvironment): Promise<void> {
+    const projectApiUrl = `${process.env.VERCEL_BASE_URL}/v9/projects`;
+    const target = ['production', 'preview', 'development'];
+    const type = 'encrypted';
+
+    const apiKeys = [
+      {
+        target,
+        type,
+        value: clientKey,
+        key: 'NOVU_CLIENT_APP_ID',
+      },
+      {
+        target,
+        type,
+        value: privateKey,
+        key: 'NOVU_API_SECRET',
+      },
+    ];
+
+    await Promise.all(
+      projectIds.map((projectId) =>
+        lastValueFrom(
+          this.httpService.post(`${projectApiUrl}/${projectId}/env${teamId ? `?teamId=${teamId}` : ''}`, apiKeys, {
+            headers: {
+              Authorization: `Bearer ${token}`,
+              'Content-Type': 'application/json',
+            },
+          })
+        )
+      )
+    );
+  }
+
+  private async getVercelProjects(accessToken: string, teamId: string | null, projectIds: string[]) {
+    const response = await lastValueFrom(
+      this.httpService.get(`${process.env.VERCEL_BASE_URL}/v4/projects${teamId ? `?teamId=${teamId}` : ''}`, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      })
+    );
+
+    return this.mapProjects(response.data.projects, projectIds);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private mapProjects(projects: any[], projectIds: string[]) {
+    return projects.reduce<MappedProjectData>(
+      (acc, curr) => {
+        const id = curr.id;
```

Comment:
retrieve the project list from the vercel and get the particular ids for the Novu env keys, as we need those env Id to update the value